### PR TITLE
fix: [CDS-77081]: Implement expiry time for failed polling subscription trigger status

### PR DESCRIPTION
**Pull request by vcalasansh**
## Describe your changes
Currently, we have an issue with polling (artifact, manifest) triggers status, in which if there is some intermittent failure in one polling attempt, we set the trigger status to `failed`, and this status is stuck in `failed` state until a new tag/version is collected.

This happens because delegate communicates with ng-manager in only 2 cases:
- When new tags are found (at least one tag which was not in delegate's cache).
- Whenever there is a failure in polling.

So if there is one intermittent failure in delegate to list tags in customer's repo, the trigger may be stuck with `failed` status for a long time, even though the failure was only intermittent, and the subsequent polling attempts were successful.

This PR fixes this issue by implementing an expiry period for `pollingSubscriptionStatus`  failed states. This relies on the fact that, if polling is consistently failing, we keep getting failure responses periodically. If this is not the case (one-off failure), we can set the `pollingSubscriptionStatus` back to `success`.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54254)

### CODE REVIEW

**Review by rathodmeetsatish — APPROVED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/webhook/polling/PollingResponseHandler.java [hunk 1]
```diff
@@ -159,7 +162,7 @@ private void handleSuccessResponse(PollingDocument pollingDocument, PollingRespo
     if (pollingDocument.getFailedAttempts() > 0) {
       pollingService.updateFailedAttempts(accountId, pollDocId, 0);
       pollingService.updateTriggerPollingStatus(
-          pollingDocument.getAccountId(), pollingDocument.getSignatures(), true, null, Collections.emptyList());
+          pollingDocument.getAccountId(), pollingDocument.getSignatures(), true, null, Collections.emptyList(), null);
```
Comment (sribalij) [inline 120-ng-manager/src/main/java/io/harness/ng/webhook/polling/PollingResponseHandler.java:165]: Correct me if i am wrong, Is this also not going to update only when we have a response from delegate in those two scenarios you mentioned in the description? Trying to see how the validity is actually solving it.

**Review by sribalij — COMMENTED** [review]
Added a few comments

File: 120-ng-manager/src/main/java/io/harness/ng/webhook/polling/PollingResponseHandler.java (hunk 1 above)
Comment (sribalij) [reply 120-ng-manager/src/main/java/io/harness/ng/webhook/polling/PollingResponseHandler.java:165]: Will discuss with you offline on this.

**Review by sribalij — APPROVED** [review]

**Comment by sribalij** [comment]
trigger review

**Review by Fernando-Dourado — APPROVED** [review]

**Comment by vcalasansh** [comment]
trigger review

**Pull request merged**